octopus/src/main/scala/octopus/App.scala

```diff
@@ -0,0 +1,33 @@
+package octopus
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait App[M[_]] {
+  def pure[A](a: A): M[A]
+  def map2[A, B, C](first: M[A], second: M[B])(combine: (A, B) => C): M[C]
+  def recover[A, B <: A](app: M[A], f: Throwable => B): M[A]
+  def map[A, B](fa: M[A])(f: A => B): M[B] = {
```

Comment:
Not sure about default implementation here. For most of the instances (or all?) you will have efficient way to implement `.map`, not relying on `map2` + `pure` combo, which is not enforced by default. I suggest to remove default here and be sure we implement `.map` in concrete instance.

Reply:
Done